desc/optimize/_constraint_wrappers.py

```diff
@@ -681,6 +681,13 @@ def unpack_state(self, x, per_objective=True):
             if t is self._eq:
                 xi_splits = np.cumsum([self._eq.dimensions[arg] for arg in self._args])
                 p = {arg: xis for arg, xis in zip(self._args, jnp.split(xi, xi_splits))}
+                p.update(  # add in dummy values for missing parameters
```

Comment:
This is because we need `params` to have the same pytree structure as `thing.params_dict`, since the logic in `FixParameter` assumes the params are flattened into the same order as `tree_leaves(thing.params_dict)`. If there are "missing leaves" from the pytree this flattened logic won't work, even though those leaves aren't actually used. `ProximalProjection` removes these params so we have to add dummy values back in.